enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+    // CloudControllerManager determines whether or not an external Cloud Controller Manager is expected to 
+    // be presented in the cluster.
+    // For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When enabled, the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+    // When omitted or disabled, no `cloud-provider` flag should be set.
+    // +kubebuilder:validation:Enum=Enabled;Disabled
+    // +optional
+    CloudControllerManager CloudControllerManagerMode `json:"cloudControllerManager,omitempty"`
+}
+
+// ExternalPlatformStatus holds the current status of the generic External infrastructure provider.
+type ExternalPlatformStatus struct{}
+```
+
+```go
+type PlatformSpec struct {
+    ...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformSpec `json:"external,omitempty"`
+}
+
+type PlatformStatus struct {
+	...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformStatus `json:"external,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create support burden. However, using of such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signals about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle hardened.
+Changing it with adding new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control. 
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
+
+- Should we invest in preparing workflows that will perform UPI with the "None"/"External" platform types installation
+  on the AWS or GCP, or existed vSphere-based workflows would be enough?
+
+### Test Plan
+
+During the first stages, we need to ensure that all OpenShift built-in operators treat the "External" platform type the same way as "None".
```

Comment:
`All` sounds like a lot, we will need to enumerate these so that we have a complete list to update when we make library-go changes